[WPE] Use drag implementation from GTK port

fujii opened the pull request:
#### da16f1faddbe958632ce195c78f8a2da795fef3d
<pre>
[WPE] Use drag implementation from GTK port
<a href="https://bugs.webkit.org/show_bug.cgi?id=294144">https://bugs.webkit.org/show_bug.cgi?id=294144</a>

Reviewed by Carlos Garcia Campos.

Extract common logic into *GLib.cpp files and start using them in WPE and GTK ports.
The implementation is mostly the same as was in GTK port. A few platform-specific
bits are left in corresponding *WPE.cpp and *GTK.cpp files.

Co-authored-by: Fujii Hironori &lt;fujii@igalia.com&gt;

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:
* LayoutTests/platform/wpe/fast/lists/drag-into-marker-expected.txt:
* LayoutTests/platform/wpe/fast/text/atsui-rtl-override-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/horizontal-bt-replaced-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/japanese-lr-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/japanese-rl-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/vertical-lr-replaced-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/vertical-lr-replaced-selection-mixed-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/vertical-rl-replaced-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/vertical-rl-replaced-selection-mixed-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-pseudo/text-selection-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/editing/dnd/datastore/datatransfer-constructor-001-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/selection-pointer-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/selection/contenteditable/modifying-selection-with-primary-mouse-button.tentative-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/selection/select-end-of-line-image.tentative-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/shadow-dom/focus/text-selection-with-delegatesFocus-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/uievents/click/click_event_target_child_parent-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-squeeze-1-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-squeeze-2-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-squeeze-3-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-squeeze-4-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-stretch-1-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-stretch-2-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-stretch-3-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-stretch-4-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-squeeze-1-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-squeeze-2-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-squeeze-3-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-squeeze-4-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-stretch-1-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-stretch-2-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-stretch-4-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-x-list-1-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/text/select-x-list-2-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/text/select-x-list-3-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/text/select-x-list-4-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/text/select-x-list-with-tspans-1-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-x-list-with-tspans-2-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-x-list-with-tspans-3-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-x-list-with-tspans-4-expected.txt:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/page/glib/DragControllerGLib.cpp: Renamed from Source/WebCore/page/gtk/DragControllerGtk.cpp.
* Source/WebCore/platform/DragData.h:
* Source/WebCore/platform/DragImage.cpp:
(WebCore::createDragImageForColor):
* Source/WebCore/platform/Pasteboard.h:
* Source/WebCore/platform/glib/DragDataGLib.cpp: Renamed from Source/WebCore/platform/gtk/DragDataGtk.cpp.
* Source/WebCore/platform/glib/PasteboardGLib.cpp: Copied from Source/WebCore/platform/gtk/PasteboardGtk.cpp.
(WebCore::pasteboardCustomDataType):
(WebCore::Pasteboard::createForCopyAndPaste):
(WebCore::Pasteboard::createForDragAndDrop):
(WebCore::Pasteboard::create):
(WebCore::Pasteboard::Pasteboard):
(WebCore::Pasteboard::selectionData const):
(WebCore::selectionDataTypeFromHTMLClipboardType):
(WebCore::Pasteboard::writeString):
(WebCore::Pasteboard::writePlainText):
(WebCore::Pasteboard::write):
(WebCore::Pasteboard::writeTrustworthyWebURLsPboardType):
(WebCore::Pasteboard::clear):
(WebCore::Pasteboard::setDragImage):
(WebCore::Pasteboard::read):
(WebCore::Pasteboard::hasData):
(WebCore::Pasteboard::typesSafeForBindings):
(WebCore::Pasteboard::typesForLegacyUnsafeBindings):
(WebCore::Pasteboard::readOrigin):
(WebCore::Pasteboard::readString):
(WebCore::Pasteboard::readStringInCustomData):
(WebCore::Pasteboard::fileContentState):
(WebCore::Pasteboard::writeMarkup):
(WebCore::Pasteboard::writeCustomData):
(WebCore::Pasteboard::changeCount const):
* Source/WebCore/platform/gtk/PasteboardGtk.cpp:
(): Deleted.
(WebCore::Pasteboard::createForCopyAndPaste): Deleted.
(WebCore::Pasteboard::createForDragAndDrop): Deleted.
(WebCore::Pasteboard::create): Deleted.
(WebCore::Pasteboard::Pasteboard): Deleted.
(WebCore::Pasteboard::selectionData const): Deleted.
(WebCore::selectionDataTypeFromHTMLClipboardType): Deleted.
(WebCore::Pasteboard::writeString): Deleted.
(WebCore::Pasteboard::writePlainText): Deleted.
(WebCore::Pasteboard::write): Deleted.
(WebCore::Pasteboard::writeTrustworthyWebURLsPboardType): Deleted.
(WebCore::Pasteboard::clear): Deleted.
(WebCore::Pasteboard::setDragImage): Deleted.
(WebCore::Pasteboard::hasData): Deleted.
(WebCore::Pasteboard::typesSafeForBindings): Deleted.
(WebCore::Pasteboard::typesForLegacyUnsafeBindings): Deleted.
(WebCore::Pasteboard::readOrigin): Deleted.
(WebCore::Pasteboard::readString): Deleted.
(WebCore::Pasteboard::readStringInCustomData): Deleted.
(WebCore::Pasteboard::fileContentState): Deleted.
(WebCore::Pasteboard::writeMarkup): Deleted.
(WebCore::Pasteboard::writeCustomData): Deleted.
(WebCore::Pasteboard::changeCount const): Deleted.
* Source/WebCore/platform/wpe/PasteboardWPE.cpp:
(WebCore::Pasteboard::createForCopyAndPaste): Deleted.
(WebCore::Pasteboard::Pasteboard): Deleted.
(WebCore::Pasteboard::writeString): Deleted.
(WebCore::Pasteboard::writePlainText): Deleted.
(WebCore::Pasteboard::write): Deleted.
(WebCore::Pasteboard::writeTrustworthyWebURLsPboardType): Deleted.
(WebCore::Pasteboard::clear): Deleted.
(WebCore::Pasteboard::hasData): Deleted.
(WebCore::Pasteboard::typesSafeForBindings): Deleted.
(WebCore::Pasteboard::typesForLegacyUnsafeBindings): Deleted.
(WebCore::Pasteboard::readOrigin): Deleted.
(WebCore::Pasteboard::readString): Deleted.
(WebCore::Pasteboard::readStringInCustomData): Deleted.
(WebCore::Pasteboard::fileContentState): Deleted.
(WebCore::Pasteboard::writeMarkup): Deleted.
(WebCore::Pasteboard::writeCustomData): Deleted.
(WebCore::Pasteboard::changeCount const): Deleted.
* Source/WebKit/Shared/wpe/WebEventFactoryWPE.cpp:
(WebKit::buttonForWPEModifiers):
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::startDrag):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebDragClient.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/wpe/WebDragClientWPE.cpp: Copied from Source/WebKit/WebProcess/WebCoreSupport/WebDragClient.cpp.
(WebKit::WebDragClient::didConcludeEditDrag):
(WebKit::WebDragClient::startDrag):
* Source/cmake/OptionsWPE.cmake:

Canonical link: <a href="https://commits.webkit.org/297961@main">https://commits.webkit.org/297961@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b8c0a6f02233e9c493fd159b20b9cdb701623b99

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/113498 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/33211 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/23650 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/119678 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/64263 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/115394 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/33820 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/41780 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/86353 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/41436 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/116446 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/27012 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/102033 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/66682 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/26279 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/20153 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/63405 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/105970 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/96395 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/20231 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/122912 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/112062 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/40511 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/30272 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/95212 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/40902 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/98237 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/94958 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/40076 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/17894 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/36674 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18234 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/40395 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/45896 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/136278 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/40053 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36551 "Found 1 new JSC stress test failure: stress/arrowfunction-constructor.js.default (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/43366 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/41810 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos reviewed (approved):
Thank you both. Wait for EWS before landing, please.

yury-s reviewed (approved):
Thanks for finishing this up! One thing I wanted to check before unconditionally (without ENABLE_DRAG_SUPPORT guards) enabling the logic in WPE and I think it is still worth doing, is if it breaks any existing scenarios where previously there were mouse down/move events and now they are converted into drag events in the page with empty delegates. It might lead to some unexpected change in behavior when there is no visual indication of the drag start etc. But on the other hand, the tests are passing which is reassuring.

yury-s committed:
[WPE] Use drag implementation from GTK port
https://bugs.webkit.org/show_bug.cgi?id=294144

Reviewed by Carlos Garcia Campos.

Extract common logic into *GLib.cpp files and start using them in WPE and GTK ports.
The implementation is mostly the same as was in GTK port. A few platform-specific
bits are left in corresponding *WPE.cpp and *GTK.cpp files.

Co-authored-by: Fujii Hironori <fujii@igalia.com>

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/wpe/TestExpectations:
* LayoutTests/platform/wpe/fast/lists/drag-into-marker-expected.txt:
* LayoutTests/platform/wpe/fast/text/atsui-rtl-override-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/horizontal-bt-replaced-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/japanese-lr-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/japanese-rl-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/vertical-lr-replaced-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/vertical-lr-replaced-selection-mixed-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/vertical-rl-replaced-selection-expected.txt:
* LayoutTests/platform/wpe/fast/writing-mode/vertical-rl-replaced-selection-mixed-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-pseudo/text-selection-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/editing/dnd/datastore/datatransfer-constructor-001-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/semantics/forms/the-input-element/selection-pointer-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/selection/contenteditable/modifying-selection-with-primary-mouse-button.tentative-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/selection/select-end-of-line-image.tentative-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/shadow-dom/focus/text-selection-with-delegatesFocus-expected.txt: Removed.
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/uievents/click/click_event_target_child_parent-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-squeeze-1-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-squeeze-2-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-squeeze-3-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-squeeze-4-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-stretch-1-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-stretch-2-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-stretch-3-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacing-stretch-4-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-squeeze-1-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-squeeze-2-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-squeeze-3-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-squeeze-4-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-stretch-1-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-stretch-2-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-textLength-spacingAndGlyphs-stretch-4-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-x-list-1-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/text/select-x-list-2-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/text/select-x-list-3-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/text/select-x-list-4-expected.txt: Removed.
* LayoutTests/platform/wpe/svg/text/select-x-list-with-tspans-1-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-x-list-with-tspans-2-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-x-list-with-tspans-3-expected.txt:
* LayoutTests/platform/wpe/svg/text/select-x-list-with-tspans-4-expected.txt:
* Source/WebCore/SourcesGTK.txt:
* Source/WebCore/SourcesWPE.txt:
* Source/WebCore/page/glib/DragControllerGLib.cpp: Renamed from Source/WebCore/page/gtk/DragControllerGtk.cpp.
* Source/WebCore/platform/DragData.h:
* Source/WebCore/platform/DragImage.cpp:
(WebCore::createDragImageForColor):
* Source/WebCore/platform/Pasteboard.h:
* Source/WebCore/platform/glib/DragDataGLib.cpp: Renamed from Source/WebCore/platform/gtk/DragDataGtk.cpp.
* Source/WebCore/platform/glib/PasteboardGLib.cpp: Copied from Source/WebCore/platform/gtk/PasteboardGtk.cpp.
(WebCore::pasteboardCustomDataType):
(WebCore::Pasteboard::createForCopyAndPaste):
(WebCore::Pasteboard::createForDragAndDrop):
(WebCore::Pasteboard::create):
(WebCore::Pasteboard::Pasteboard):
(WebCore::Pasteboard::selectionData const):
(WebCore::selectionDataTypeFromHTMLClipboardType):
(WebCore::Pasteboard::writeString):
(WebCore::Pasteboard::writePlainText):
(WebCore::Pasteboard::write):
(WebCore::Pasteboard::writeTrustworthyWebURLsPboardType):
(WebCore::Pasteboard::clear):
(WebCore::Pasteboard::setDragImage):
(WebCore::Pasteboard::read):
(WebCore::Pasteboard::hasData):
(WebCore::Pasteboard::typesSafeForBindings):
(WebCore::Pasteboard::typesForLegacyUnsafeBindings):
(WebCore::Pasteboard::readOrigin):
(WebCore::Pasteboard::readString):
(WebCore::Pasteboard::readStringInCustomData):
(WebCore::Pasteboard::fileContentState):
(WebCore::Pasteboard::writeMarkup):
(WebCore::Pasteboard::writeCustomData):
(WebCore::Pasteboard::changeCount const):
* Source/WebCore/platform/gtk/PasteboardGtk.cpp:
(): Deleted.
(WebCore::Pasteboard::createForCopyAndPaste): Deleted.
(WebCore::Pasteboard::createForDragAndDrop): Deleted.
(WebCore::Pasteboard::create): Deleted.
(WebCore::Pasteboard::Pasteboard): Deleted.
(WebCore::Pasteboard::selectionData const): Deleted.
(WebCore::selectionDataTypeFromHTMLClipboardType): Deleted.
(WebCore::Pasteboard::writeString): Deleted.
(WebCore::Pasteboard::writePlainText): Deleted.
(WebCore::Pasteboard::write): Deleted.
(WebCore::Pasteboard::writeTrustworthyWebURLsPboardType): Deleted.
(WebCore::Pasteboard::clear): Deleted.
(WebCore::Pasteboard::setDragImage): Deleted.
(WebCore::Pasteboard::hasData): Deleted.
(WebCore::Pasteboard::typesSafeForBindings): Deleted.
(WebCore::Pasteboard::typesForLegacyUnsafeBindings): Deleted.
(WebCore::Pasteboard::readOrigin): Deleted.
(WebCore::Pasteboard::readString): Deleted.
(WebCore::Pasteboard::readStringInCustomData): Deleted.
(WebCore::Pasteboard::fileContentState): Deleted.
(WebCore::Pasteboard::writeMarkup): Deleted.
(WebCore::Pasteboard::writeCustomData): Deleted.
(WebCore::Pasteboard::changeCount const): Deleted.
* Source/WebCore/platform/wpe/PasteboardWPE.cpp:
(WebCore::Pasteboard::createForCopyAndPaste): Deleted.
(WebCore::Pasteboard::Pasteboard): Deleted.
(WebCore::Pasteboard::writeString): Deleted.
(WebCore::Pasteboard::writePlainText): Deleted.
(WebCore::Pasteboard::write): Deleted.
(WebCore::Pasteboard::writeTrustworthyWebURLsPboardType): Deleted.
(WebCore::Pasteboard::clear): Deleted.
(WebCore::Pasteboard::hasData): Deleted.
(WebCore::Pasteboard::typesSafeForBindings): Deleted.
(WebCore::Pasteboard::typesForLegacyUnsafeBindings): Deleted.
(WebCore::Pasteboard::readOrigin): Deleted.
(WebCore::Pasteboard::readString): Deleted.
(WebCore::Pasteboard::readStringInCustomData): Deleted.
(WebCore::Pasteboard::fileContentState): Deleted.
(WebCore::Pasteboard::writeMarkup): Deleted.
(WebCore::Pasteboard::writeCustomData): Deleted.
(WebCore::Pasteboard::changeCount const): Deleted.
* Source/WebKit/Shared/wpe/WebEventFactoryWPE.cpp:
(WebKit::buttonForWPEModifiers):
(WebKit::WebEventFactory::createWebMouseEvent):
* Source/WebKit/SourcesWPE.txt:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::startDrag):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebProcess/WebCoreSupport/WebDragClient.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/wpe/WebDragClientWPE.cpp: Copied from Source/WebKit/WebProcess/WebCoreSupport/WebDragClient.cpp.
(WebKit::WebDragClient::didConcludeEditDrag):
(WebKit::WebDragClient::startDrag):
* Source/cmake/OptionsWPE.cmake:

Canonical link: https://commits.webkit.org/297961@main
